File: basic_samples/Complex_Dynamic/python/test_sample.py

import json
import requests
import traceback

import sample as sample
import sendOMF as sendOMF
import omfHelper as omfHelper
import auth as auth

config = {}


def suppressError(call):
    try:
        call()
    except Exception as e:
        print(f'Encountered Error: {e}')


def checkData():
    if config['destinationOCS']:
        checkLastOCSVal()
    # don't have to check others as they are sync and we get instant feedback on success from the app itself


def checkLastOCSVal():
    msg_headers = {
        "Authorization": auth.getAuthHeader()
    }
    url = config['omfURL'].split('/omf')[0] + \
        '/streams/Tank1Measurements/data/last'
    response = requests.get(
        url,
        headers=msg_headers,
        verify=config['verify']
    )

    # response code in 200s if the request was successful!
    if response.status_code < 200 or response.status_code >= 300:
        print(msg_headers)
        response.close()
        print(
            f'Response from was bad.  message: {response.status_code} {response.text}.')
        print()
        raise Exception(
            f'OMF message was unsuccessful,  {response.status_code}:{response.text}')


def sendTypeDelete(_type):
    sendOMF.sendCall(json.dumps(_type), 'type', 'delete')


def sendContainerDelete(container):
    sendOMF.sendCall(json.dumps(container), 'container', 'delete')


def test_main():
    global config
    # Tests to make sure the sample runs as expected

    try:
        config = sample.main(True, ['2,3', 'n'])
        checkData()

    except Exception as ex:
        print(f'Encountered Error: {ex}.')
        print
        traceback.print_exc()
        print
        success = False
        raise ex

    finally:
        print('Deletes')
        print

        suppressError(lambda: sendContainerDelete(omfHelper.getContainer()))
        suppressError(lambda: sendTypeDelete(omfHelper.getType()))


if __name__ == '__main__':
    test_main()

File: basic_samples/Complex_Dynamic/nodeJS/Test_Sample.js
var config = require("./config.js");
const readline = require("readline");
var authObj = require("./auth.js");
var omfObj = require("./omfClient.js");
var sample = require("./Sample.js");

var logError = function(err) {
  success = false;
  errorCap = err;

  console.log("Error");
  console.trace();
  console.log(err.message);
  console.log(err.stack);
  console.log(err.options.headers["Operation-Id"]);
  throw err;
};

sample
  .app(["1,2", "n"])
  .then(function(res) {
    let count = 1;
    function checkFlag() {
      if (count > 50) {
        throw new Error(
          "Main app didn't finish quickly enough.  chacnes are it is hung somewhere..."
        );
      }
      if (global.ending == false) {
        setTimeout(checkFlag, 100);
        count++;
      } else {
        checkData(res).then(
          deleteContainer(res).then(deleteType(res).then(checkDone(res)))
        );
      }
    }

    checkFlag();
  })
  .catch(function(err) {
    logError(err);
  });

checkData = function(res) {
  console.log(res);
  if (global.config.OCS) {
    var restCall = require("request-promise");
    console.log("res2");

    url =
      global.config.omfURL.split("/omf")[0] +
      "/streams/Tank1Measurements/data/last";
    return restCall({
      url: url,
      method: "Get",
      headers: { Authorization: "bearer " + global.authClient.token }
    }).then(function(res) {
      console.log(res);
      return new Promise(function(resolve, reject) {
        resolve();
      });
    });
  } else {
    return new Promise(function(resolve, reject) {
      resolve();
    });
  }
};

deleteContainer = function(res) {
  console.log(res);
  console.log("Deleting Container");
  containerObj = sample.omfContainer();
  if (global.authClient.tokenExpires >= sample.nowSeconds) {
    return function(res) {
      refreshToken(res, authClient);
      return omfClient.deleteContainer(containerObj);
    };
  } else {
    return omfClient.deleteContainer(containerObj);
  }
};
done = false;

deleteType = function(res) {
  console.log(res);
  console.log("Delete Type");
  if (authClient.tokenExpires >= sample.nowSeconds) {
    return function(res) {
      refreshToken(res, authClient);
      return omfClient.deleteType(sample.omfType).then(process.exit(0));
    };
  } else {
    return omfClient.deleteType(sample.omfType).then(process.exit(0));
  }
};
